04-spatial-operations.Rmd

```diff
@@ -217,8 +222,16 @@ Since mid-2017 (with the release of **sf** `0.5-3`) the base R function `aggrega
 Building on the example presented the previous section (\@ref(spatial-subsetting)), we demonstrate this by aggregating the population of countries that intersect with the buffer represented by the circular `buff` object created in the previous section.
 
 ```{r}
-buff_agg = aggregate(x = africa["pop"], by = buff, FUN = sum)
+buff_agg = aggregate(x = africa[, "pop"], by = buff, FUN = sum)
+```
+<!--
+show also tidyverse way, so what you are doing is basically a spatial join and a subsequent aggregation without a grouping variable. Didactically, it might be better to present a grouping variable.
+```{r}
+st_join(buff, africa[, "pop"]) %>%
```

Comment:
note that this is the tidyverse version of the aggregate command. So basically, you are doing a spatial join and then summarize all data without a grouping variable...

Reply:
That's one way of putting it. Spatial aggregation is a type of spatial join.

Reply:
Well, I would say you selected a 'special' example. The `aggregate` function actually dissolves polygons. However, your input `buff` was only one polygon, so there was nothing to dissolve. 
Check this out:
```r
plot(aggregate(africa[, "pop"], by = list(africa$subregion), FUN = sum, na.rm = TRUE))
# which is the same as:
plot(group_by(africa, subregion) %>% summarize(pop = sum(pop, na.rm = TRUE)))
```
See, what I mean. 
And yes, you could also give a spatial object to the by command, and what you get back, is the aggregate of the intersection. So this might be a bit confusing, which is I would skip this, and stick with a real dissolving example (like here the africa example or the us_states examples). We can show first how to do it in base R (`aggregate`) and then by using `summarize()`

Reply:
Perhaps the example should be Chapter 3 as the operation does not change the geometry of either object. I think I know what you mean... The example is deliberately simple, to show how the function `aggregate` works. I think a second example should build on this one on the basis that you should start simple and add complexity. Suggestions along those lines welcome e.g. via updates to this PR.

Reply:
It does change the geometry, see the polygon boundaries (just 5 polygons left), and compare them to `plot(africa[, "pop"])` (51 polygons). In fact, that's the whole point of dissolving. Somehow, I have the impression we talk at cross-purposes... sorry, if that's the case!

Reply:
Just for comparison purposes:
```r
par(mfrow = c(1, 2))
plot(africa[, "pop"])
group_by(africa, subregion) %>% 
  summarize(pop = sum(pop, na.rm = TRUE)) %>%
  select(pop) %>% 
  plot
```

Reply:
Yes we were talking about different things: I was talking about the example I added : )

Your example clearly changes the geometry. I was referring to the buffer example when I said it could go in Chapter 3. What do you think?

Reply:
Ah, ok, then this is resolved 😄 . 
If we are talking about `aggregate(x = africa[, "pop"], by = buff, FUN = sum)`, I wouldn't put it into c3, since it is doing a spatial join (using st_intersect in the background), so I would put it to the spatial join subsection. I will be off now helping a friend with painting his new house! So have fun and read and hear you soon!
BTW, I have also some loose thoughts how to structure the book, will put them onto paper within the next few days, and give this to you as a basis for discussion.